src/lib/routing/router.ts

```diff
@@ -20,21 +20,53 @@ const defaultErrorHandler = (auto405) => async (err: Error, req: EReq, res: ERes
   res.withStatus(500).json({ error: err.message });
 }
 
+/**
+ * Handles preflight requests from trusted origins configured by the user when initializing a router.
+ * Note that the wildcard value "*" will fail if the request is sent with credentials (see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives)
+ * @param autoCorsPreflight the object containing CORS preflight option of trustedOrigins, an array of trusted origins.
+ * @returns 200 if the preflight request succeeds, 403 if it fails
+ */
+const preflightHandler = (autoCorsPreflight: AutoCorsPreflightOptions) => async (req: EReq, res: ERes) => {
+  if (autoCorsPreflight.trustedOrigins.length === 0) {
+    return res.sendStatus(403);
+  }
+  let originHeaderValue: string | null = null;
+  if (autoCorsPreflight.trustedOrigins.length === 1 && autoCorsPreflight.trustedOrigins[0] === '*') {
+    originHeaderValue = '*';
+  } else {
+    const origin = req.headers.get("origin").toLowerCase();
+    if (autoCorsPreflight.trustedOrigins.some((trustedOrigin) => trustedOrigin.toLowerCase() === origin)) {
+      originHeaderValue = origin;
+    }
+  }
+  if (originHeaderValue == null) {
+    return res.sendStatus(403);
+  }
+  res.set({
+    'access-control-allow-origin': originHeaderValue,
+    'access-control-allow-methods': req.headers.get("access-control-request-method"),
+  })
+  return res.sendStatus(200);
+}
+
 export class Router {
   requestHandlers: Array<RequestHandler> = [];
   errorHandlers: Array<ErrorMiddleware> = [];
   config: EConfig = {
     parseCookie: true,
     auto405: true,
     extractRequestParameters: true,
-    autoContentType: false
+    autoContentType: false,
   };
 
   constructor(config?: EConfig) {
     this.config = {
       ...this.config,
       ...config
     }
+    if (!this.config.autoCorsPreflight) {
+      this.options("*", preflightHandler(this.config.autoCorsPreflight));
+    }
```

Comment:
This only runs if there *isn't* an `autoCorsPreflight` config.  I think you want to run it only if there is?

Reply:
Just realised this came about because I made this mistake originally in my earlier suggestion - sorry!